sdl-lang/src/main/scala/io/smartdatalake/meta/configexporter/ConfigJsonExporter.scala

```diff
@@ -0,0 +1,85 @@
+
+object ConfigJsonExporter extends SmartDataLakeLogger {
+
+  val appType: String = getClass.getSimpleName.replaceAll("\\$$", "") // remove $ from object name and use it as appType
+
+  protected val parser: OptionParser[ConfigJsonExporterConfig] = new OptionParser[ConfigJsonExporterConfig](appType) {
+    override def showUsageOnError: Option[Boolean] = Some(true)
+    opt[String]('c', "config")
+      .required()
+      .action((value, c) => c.copy(configPaths = value.split(',')))
+      .text("One or multiple configuration files or directories containing configuration files for SDLB, separated by comma.")
+    opt[String]('f', "filename")
+      .optional()
+      .action((value, c) => c.copy(filename = value))
+      .text("File to export configuration to. Default: exportedConfig.json")
+    opt[Boolean]("enrichOrigin")
+      .optional()
+      .action((value, c) => c.copy(enrichOrigin = value))
+      .text("Whether to add an additional property 'origin' including source filename and line number to first class configuration objects.")
+    help("help").text("Display the help text.")
+  }
+
+  /**
+   * Takes as input a SDL Config and exports it as one json document, everything resolved.
+   * Additionally a separate file with the mapping of first class config objects to source code origin is created.
+   */
+  def main(args: Array[String]): Unit = {
+    val config = ConfigJsonExporterConfig()
+    // Parse all command line arguments
+    parser.parse(args, config) match {
+      case Some(config) =>
+
+        // create json
+        val configAsJson = exportConfigJson(config)
+
+        // write file
+        logger.info(s"Writing config json to file ${config.filename}")
+        Files.write(Paths.get(config.filename), configAsJson.getBytes, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)
+
+      case None =>
+        logAndThrowException(s"Aborting ${appType} after error", new ConfigurationException("Couldn't set command line parameters correctly."))
+    }
+  }
+
+  def exportConfigJson(config: ConfigJsonExporterConfig): String = {
+    val defaultHadoopConf: Configuration = new Configuration()
+    val sdlConfig = ConfigLoader.loadConfigFromFilesystem(config.configPaths, defaultHadoopConf)
+    // remove additional config paths introduced by system properties...
+    val configKeysToRemove = sdlConfig.root.keySet().asScala.diff(Set(ConfigParser.CONFIG_SECTION_ACTIONS, ConfigParser.CONFIG_SECTION_CONNECTIONS, ConfigParser.CONFIG_SECTION_DATAOBJECTS, ConfigParser.CONFIG_SECTION_GLOBAL))
+    val reducedSdlConfig = configKeysToRemove.foldLeft(sdlConfig)((config,key) => config.withoutPath(key))
+    // enrich origin of first class config objects
+    val descriptionRegex = "(.*): ([0-9]+)(-[0-9]+)?".r
+    val enrichedSdlConfig = reducedSdlConfig.root().keySet().asScala.diff(Set(ConfigParser.CONFIG_SECTION_GLOBAL)).foldLeft(reducedSdlConfig){
```

Comment:
It's a bit hard to follow. Basically you just want to list a hardcoded list of config sections to enrich, right?
```suggestion
val configSectionsToEnrich =  Set(ConfigParser.CONFIG_SECTION_ACTIONS, ConfigParser.CONFIG_SECTION_CONNECTIONS, ConfigParser.CONFIG_SECTION_DATAOBJECTS)
    val enrichedSdlConfigToExport =configSectionsToEnrich.foldLeft(sdlConfigToExport){
```

Reply:
Good point, will do.